internal/helper/handler_context.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package helper
+
+import "context"
+
+type (
+	Caller           string
+	contextMethodKey struct{}
+	callerContext    struct {
+		ctx context.Context
+	}
+)
+
+const (
+	DataRead     Caller = "DataRead"
+	RefreshState Caller = "RefreshState"
+	CreateState  Caller = "CreateState"
+	UpdateState  Caller = "UpdateState"
+)
+
+func NewCallerContext(ctx context.Context) *callerContext {
```

Comment:
I do not think we need to create a struct 'callerContext' here.
All the helper functions are useful, but avoid the struct and the method binding

Reply:
Updated.